chemprop/v2/data/splitting.py

```diff
@@ -206,9 +207,48 @@ def _unpack_astartes_result(
         train_idxs, val_idxs, test_idxs = result[-3], result[-2], result[-1]
     else:
         train_idxs, test_idxs = result[-2], result[-1]
-    if data is None:
-        return train_idxs, val_idxs, test_idxs
-    train = [data[i] for i in train_idxs]
-    val = [data[i] for i in val_idxs]
-    test = [data[i] for i in test_idxs]
+    return train_idxs, val_idxs, test_idxs
+
+
+def split_monocomponent(
+    datapoints: Sequence[MoleculeDatapoint], *args, **kwargs
+) -> tuple[list[MoleculeDatapoint], ...] | tuple[list[list[MoleculeDatapoint]], ...]:
+    """Splits monocomponent data into training, validation, and test splits."""
+
+    # split the data
+    train_idxs, val_idxs, test_idxs = split_data(datapoints, *args, **kwargs)
+
+    if isinstance(train_idxs[0], list):
+        # convert indices to datapoints for each fold
+        train = [[datapoints[i] for i in fold] for fold in train_idxs]
+        val = [[datapoints[i] for i in fold] for fold in val_idxs]
+        test = [[datapoints[i] for i in fold] for fold in test_idxs]
+    else:
+        # convert indices to datapoints
+        train = [datapoints[i] for i in train_idxs]
+        val = [datapoints[i] for i in val_idxs]
+        test = [datapoints[i] for i in test_idxs]
```

Comment:
IMO type-checking should be avoided when possible:
1. it's non-pythonic: Python doesn't have strong typing support and goes against the core principle of duck typing
2. it obscures code meaning: I interpret this `if/else` block as branching depending on the result of `split_data`, which itself is a function of the arguments passed into the function. That is, depending on whether we're doing CV or not, we will get a result of `tuple[list[list[int]], ...]` or `tuple[list[int]], ...]`. We should just check that directly here by branching based on the value of `split_type`.